pkg/graphql/request.go

```diff
@@ -41,6 +41,7 @@ type Request struct {
 	document     ast.Document
 	isParsed     bool
 	isNormalized bool
+	isValidated  bool
```

Comment:
I have only one concern here if we use a simple boolean for "validate once":
If we use the same request and validate it for 2 different schemas (e.g. a supergraph schema and a subgraph SDL) then it will only validate for the first schema.
We won't catch an error if it's not valid for the second case.

Reply:
Yeah, got it. We talked about hashing the schema and then associating the validation with the hash. Will fix.